src/components/SingleName/RecordsItem.js

```diff
@@ -148,26 +165,22 @@ const Editable = ({
                     )}
                   </Mutation>
                 </Action>
-              ) : (
-                <Action>
-                  <Pencil
-                    onClick={startEditing}
-                    data-testid={`edit-${keyName.toLowerCase()}`}
-                  />
-                </Action>
-              )}
+              ) : <Actionable startEditing={startEditing} keyName={keyName} value={value} />}
             </RecordsContent>
             {editing ? (
               <>
                 <EditRecord>
-                  <Input
-                    placeholder={getPlaceholder(type)}
-                    onChange={e => updateValue(e.target.value)}
-                    valid={isValid}
-                    invalid={isInvalid}
+                  <DetailsItemInput 
+                    newValue={newValue}
+                    dataType={type}
+                    contentType={domain.contentType}
+                    updateValue={updateValue}
+                    isValid={isValid}
+                    isInvalid={isInvalid}
                   />
                 </EditRecord>
                 <SaveCancel
+                  oldcontentWarning={ type === 'content' && domain.contentType === 'oldcontent'}
```

Comment:
we should make `<SaveCancel>` prop generic so we can pass any type of warning too it and change the message based on the component since we use SaveCancel in all components

Reply:
👍